manager/orchestrator/updater.go

```diff
@@ -302,6 +389,10 @@ func (u *Updater) isServiceDirty(service *api.Service) bool {
 		!reflect.DeepEqual(u.newService.Spec.Endpoint, service.Spec.Endpoint)
 }
 
+func (u *Updater) isInstanceDirty(instance []*api.Task) bool {
+	return len(instance) > 1 || u.isTaskDirty(instance[0])
```

Comment:
`len(instance) > 1 || (len(instance) == 1 && u.isTaskDirty(instance[0]))`
I think it would not happen in this context. But I feel it better to make sure no out of bound access.